File: examples/snippets/modules/ROOT/examples/build-flink-streamlets-scala/step0/src/main/scala/com/example/ReportPrinter.scala
package com.example

import cloudflow.flink._
import cloudflow.streamlets.StreamletShape

// TODO rename to ReportPrinter
class ReportPrinterStep0 extends FlinkStreamlet {
  // 1. TODO Create inlets and outlets
  // 2. TODO Define the shape of the streamlet
  val shape = StreamletShape.empty
  // 3. TODO Override createLogic to provide StreamletLogic
  def createLogic(): FlinkStreamletLogic = new FlinkStreamletLogic() {
    def buildExecutionGraph = ()
  }
}

File: examples/snippets/modules/ROOT/examples/build-flink-streamlets-java/step0/src/main/java/com/example/ReportPrinter.java
package com.example;

import cloudflow.streamlets.StreamletShape;
import cloudflow.flink.*;

public class ReportPrinter extends FlinkStreamlet {
  // 1. TODO Create inlets and outlets
  // 2. TODO Define the shape of the streamlet
  public StreamletShape shape() { return StreamletShape.empty(); }
  // 3. TODO Override createLogic to provide StreamletLogic
  public FlinkStreamletLogic createLogic() { 
    return new FlinkStreamletLogic(getContext()) {
      public void buildExecutionGraph() {
      }
    };
  }

}